lte/gateway/python/magma/mobilityd/tests/test_multi_apn_ip_alloc.py

```diff
@@ -0,0 +1,307 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import ipaddress
+import unittest
+
+from magma.mobilityd import subscriberdb_client
+
+from lte.protos.subscriberdb_pb2 import (
+    LTESubscription,
+    SubscriberData,
+    SubscriberState,
+    SubscriberID,
+    SubscriberUpdate,
+    Non3GPPUserProfile,
+    APNConfiguration,
+)
+
+from lte.protos.mconfig.mconfigs_pb2 import MobilityD
+from magma.mobilityd.ip_descriptor import IPDesc, IPType
+from magma.mobilityd.ip_address_man import IPAddressManager, \
+    IPNotInUseError, MappingNotFoundError
+from magma.subscriberdb.sid import SIDUtils
+
+
+class MockedSubscriberDBStub:
+    # subscriber map
+    subs = {}
+
+    def __init__(self):
+        pass
+
+    def GetSubscriberData(self, sid):
+        cls = self.__class__
+        return cls.subs.get(str(sid), None)
+
+    @classmethod
+    def add_sub(cls, sid: str, apn: str, ip: str, vlan: str):
+        sub_db_sid = SIDUtils.to_pb(sid)
+        lte = LTESubscription()
+        lte.state = LTESubscription.ACTIVE
+        state = SubscriberState()
+        state.lte_auth_next_seq = 1
+        non_3gpp = Non3GPPUserProfile()
+        subs_data = SubscriberData(sid=sub_db_sid, lte=lte, state=state, non_3gpp=non_3gpp)
+
+        cls.subs[str(sub_db_sid)] = subs_data
+        cls.add_sub_ip(sid, apn, ip, vlan)
+
+    @classmethod
+    def add_incomplete_sub(cls, sid: str):
+        sub_db_sid = SIDUtils.to_pb(sid)
+        lte = LTESubscription()
+        lte.state = LTESubscription.ACTIVE
+        state = SubscriberState()
+        state.lte_auth_next_seq = 1
+        subs_data = SubscriberData(sid=sub_db_sid, lte=lte, state=state)
+        cls.subs[str(sub_db_sid)] = subs_data
+
+    @classmethod
+    def add_sub_ip(cls, sid: str, apn: str, ip: str, vlan: str):
+        sub_db_sid = SIDUtils.to_pb(sid)
+        apn_config = APNConfiguration()
+        apn_config.context_id = 1
+        apn_config.service_selection = apn
+        apn_config.assigned_static_ip = ip
+        apn_config.vlan_id = vlan
+
+        subs_data = cls.subs[str(sub_db_sid)]
+        subs_data.non_3gpp.apn_config.extend([apn_config])
+
+    @classmethod
+    def clear_subs(cls):
+        cls.subs = {}
+
+
+class StaticIPAllocationTests(unittest.TestCase):
+    """
+    Test class for the Mobilityd Static IP Allocator
+    """
```

Comment:
Bad copypasta. This is the vlan allocator test. Rename the class too

Reply:
fixed.